libs/api/ebpf_api.cpp

```diff
@@ -385,75 +216,46 @@ ebpf_api_load_program(
         goto Done;
     }
 
-    *count_of_map_handles = 0;
-    handles = get_all_map_handles();
-    for (const auto& map_handle : handles) {
-        map_handles[*count_of_map_handles] = reinterpret_cast<HANDLE>(map_handle);
-        (*count_of_map_handles)++;
-    }
-
-    byte_code.resize(byte_code_size);
-    result = resolve_maps_in_byte_code(program_handle, byte_code);
-    if (result != ERROR_SUCCESS) {
-        goto Done;
-    }
-
-    result = resolve_ec_function(EBPF_EC_FUNCTION_LOG, &log_function_address);
-    if (result != ERROR_SUCCESS)
-        goto Done;
-
-    if (execution_type == EBPF_EXECUTION_JIT) {
-        std::map<uint32_t, uint64_t> helper_id_to_adddress;
-        result = build_helper_id_to_address_map(program_handle, byte_code, helper_id_to_adddress);
-        if (result != ERROR_SUCCESS)
+    // populate load_info.
+    load_info.file_name = const_cast<char*>(file_name);
+    load_info.section_name = const_cast<char*>(section_name);
+    load_info.program_name = nullptr;
+    load_info.program_type = program_type;
+    load_info.program_handle = program_handle;
+    load_info.execution_type = execution_type;
+    load_info.byte_code = byte_code.data();
+    load_info.byte_code_size = (uint32_t)byte_code_size;
+    load_info.execution_context = execution_context_kernel_mode;
+    load_info.map_count = (uint32_t)get_map_descriptor_size();
+
+    if (load_info.map_count > 0) {
+        load_info.handle_map = (fd_handle_map*)calloc(load_info.map_count, sizeof(fd_handle_map));
```

Comment:
This memory is leaked if this function throws an exception. Why use calloc and not a vector?

Reply:
changed to vector and added a try-catch block.